primary_beam_correction.py

```diff
@@ -132,70 +123,62 @@ def standard_deviation(data):
     while diff > 0:
         i += 1
         if i > 50:
-            return sd
+            return 1/(sd)**2
         old_sd = sd
         old_len = len(data)
         cut = np.abs(data - med) < 5.0 * sd
         if np.all(~cut):
-            return sd
+            return 1/(sd)**2
         data = data[cut]
-        sd = mad(data)[1]
+        med, sd = mad(data)
         diff = old_len - len(data)
         if sd == 0.0:
             return old_sd
     return 1/(sd)**2
 
 
-def weigted_average(arr, weights):
-    '''
+def weighted_average(arr, weights):
+    """
     Computing weighted average of all the frequency planes.
-    '''
+    """
     wt_average = np.average(arr, weights=weights, axis=0)
     return wt_average
 
 
 def primary_beam_correction(beam_pattern, path):
-    '''
+    """
     Correcting for the primary beam effects.
-    '''
+    """
     raw_image = read_fits(path)
-    raw_img = []
-    try:
-        nterm = raw_image.header['NTERM']
-    except KeyError:
-        nterm = _get_value_from_history('NTERM', raw_image.header)
-    nterm = int(nterm)
-    for j in range(nterm, raw_image.header['NSPEC']+nterm):
-        raw_img.append(np.ravel(raw_image.data[0, j, :, :]))
+    nterm = raw_image.header['NTERM']
     snr = []
     pbc_image = []
     for i in range(len((beam_pattern))):
         # Getting all the values with attenuated flux of less than 10% of the peak
         ind = np.argwhere(beam_pattern[i] <= 0.10)
         if len(ind) > 0:
             beam_pattern[i][ind] = np.nan
-        # Getting the rms noise in each image (before primary beam correction)
-        snr.append(standard_deviation(raw_img[i]))
+        # Getting the rms noise in each image (before primary beam correction)    
```

Comment:
Trailing whitespace.

You might want to run flake8 on the code again.

Reply:
Done